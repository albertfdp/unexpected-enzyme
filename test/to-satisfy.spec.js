import expect from './unexpected-enzyme';
import { mount } from 'enzyme';
import React from 'react';
import PropTypes from 'prop-types';

const User = ({ children }) => (
  <div className="user">
    <h3 id="name">Harriet</h3>
    {children}
  </div>
);

User.propTypes = {
  children: PropTypes.node
};

describe('to-satisfy', () => {
  describe('when it is a subtree', () => {
    it('passes when the actual matches the expected', () => {
      const wrapper = mount(
        <User>
          <div id="child">Children</div>
        </User>
      );

      expect(
        wrapper,
        'to satisfy',
        <div className="user">
          <h3 id="name">Harriet</h3>
          <div id="child">Children</div>
        </div>
      );
    });

    it('fails when the actual does not the expected', () => {
      const wrapper = mount(
        <User>
          <div id="child">Foo</div>
        </User>
      );

      expect(
        () =>
          expect(
            wrapper,
            'to satisfy',
            <div className="user">
              <h3 id="name">Harriet</h3>
              <div id="child">Children</div>
            </div>
          ),
        'with error matching snapshot'
      );
    });

    it('fails when the actual not matching the expected type', () => {
      const wrapper = mount(
        <User>
          <div id="child">Foo</div>
        </User>
      );

      expect(
        () => expect(wrapper, 'to satisfy', <h3>Header</h3>),
        'with error matching snapshot'
      );
    });
  });

  describe('with the exhaustively flag', () => {
    it('passes if the wrapper represents an element that is equal to the given element', () =>
      expect(
        mount(
          <User>
            <div id="child">Children</div>
          </User>
        ),
        'to exhaustively satisfy',
        <div className="user">
          <h3 id="name">Harriet</h3>
          <div id="child">Children</div>
        </div>
      ));

    it('fails when the actual does not match the expected', () =>
      expect(
        () =>
          expect(
            mount(
              <User>
                <div id="child">Children</div>
              </User>
            ),
            'to exhaustively satisfy',
            <div id="kids">Kids</div>
          ),
        'with error matching snapshot'
      ));

    describe('when negated', () => {
      it('passes when actual is not equal the expected', () =>
        expect(
          mount(
            <User>
              <div id="child">Children</div>
            </User>
          ),
          'not to exhaustively satisfy',
          <div className="user">
            <h3 id="name">Harriet</h3>
            <div id="kids">Kids</div>
          </div>
        ));

      it('fails when actual is equal the expected', () =>
        expect(
          () =>
            expect(
              mount(
                <User>
                  <div id="child">Children</div>
                </User>
              ),
              'not to exhaustively satisfy',
              <div className="user">
                <h3 id="name">Harriet</h3>
                <div id="child">Children</div>
              </div>
            ),
          'with error matching snapshot'
        ));
    });
  });
});
