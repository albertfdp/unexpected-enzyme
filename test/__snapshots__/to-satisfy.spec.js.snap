// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to-satisfy when it is a subtree fails when the actual does not the expected 1`] = `
"expected
<div class=\\"user\\">
  <h3 id=\\"name\\">Harriet</h3>
  <div id=\\"child\\">Foo</div>
</div>
to satisfy <User><div id=\\"child\\">Children</div></User>

<User>
  <div id=\\"child\\">
    Foo // -Foo
        // +Children
  </div>
</User>"
`;

exports[`to-satisfy when it is a subtree fails when the actual not matching the expected type 1`] = `
"expected
<div class=\\"user\\">
  <h3 id=\\"name\\">Harriet</h3>
  <div id=\\"child\\">Foo</div>
</div>
to satisfy <h3>Header</h3>

<User // should be <h3
>
  <div id=\\"child\\"> // wrapper should be removed
    Foo // -Foo
        // +Header
  </div> // wrapper should be removed
</User>"
`;

exports[`to-satisfy with the exhaustively flag fails when the actual does not match the expected 1`] = `
"expected
<div class=\\"user\\">
  <h3 id=\\"name\\">Harriet</h3>
  <div id=\\"child\\">Children</div>
</div>
to exhaustively satisfy <User><div id=\\"kids\\">Kids</div></User>

<User>
  <div id=\\"child\\" // expected 'child' to equal 'kids'
                  //
                  // -child
                  // +kids
  >
    Children // -Children
             // +Kids
  </div>
</User>"
`;

exports[`to-satisfy with the exhaustively flag when negated fails when actual is equal the expected 1`] = `
"expected
<div class=\\"user\\">
  <h3 id=\\"name\\">Harriet</h3>
  <div id=\\"child\\">Children</div>
</div>
not to exhaustively satisfy <User><div id=\\"child\\">Children</div></User>"
`;
